File: bumperbot_cpp_examples/src/simple_publisher.cpp
#include<rclcpp/rclcpp.hpp>
#include<std_msgs/msg/string.hpp>
#include<chrono>

using namespace std::chrono_literals;
class SimplePublisher : public rclcpp::Node
{
    public:
        SimplePublisher() : Node("simple_publisher"), counter_(0)
        {
            pub_ = create_publisher<std_msgs::msg::String>("chatter",10);
            timer_ = create_wall_timer(1s, std::bind(&SimplePublisher::timerCallback, this)); 

            RCLCPP_INFO(get_logger(), "Simple Publisher has been started.");
        }
    private:
        unsigned int counter_;
        rclcpp::Publisher<std_msgs::msg::String>::SharedPtr pub_;
        rclcpp::TimerBase::SharedPtr timer_;


        void timerCallback()
        {
            auto msg = std_msgs::msg::String();
            msg.data = "Hello Ayush: " + std::to_string(counter_++);
            RCLCPP_INFO(get_logger(), "Publishing: '%s'", msg.data.c_str());
            pub_->publish(msg);
        }
};

int main(int argc, char *argv[])
{
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<SimplePublisher>());
    rclcpp::spin(std::make_shared<SimplePublisher>());
    rclcpp::shutdown();
    return 0;
}

File: bumperbot_py_examples/bumperbot_py_examples/simple_publisher.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String

class SimplePublisher(Node):
    def __init__(self):
        super().__init__("simplle_publisher")
        
        # Create the publisher
        self.pub = self.create_publisher(String, "chatter", 10)
        
        # Initialize the counter and frequency
        self.counter_ = 0
        self.frequency_ = 1.0
        
        # Log the publishing frequency
        self.get_logger().info(f"Publishing at {self.frequency_} Hz")
        
        # Create the timer and specify the callback
        self.timer_ = self.create_timer(self.frequency_, self.timerCallback)

    def timerCallback(self):
        # Create a new message
        msg = String()
        msg.data = f"Hello Ayush, counter: {self.counter_}"
        
        # Publish the message
        self.pub.publish(msg)
        
        # Increment the counter
        self.counter_ += 1

def main():
    # Initialize the ROS 2 client library
    rclpy.init()
    
    # Create the SimplePublisher node
    simple_publisher = SimplePublisher()
    
    # Spin to keep the program running and processing callbacks
    rclpy.spin(simple_publisher)
    
    # Clean up the node and shutdown ROS 2
    simple_publisher.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()
